lib/guardian.ex

```diff
@@ -278,12 +278,14 @@ defmodule Guardian do
     else
       params
     end
-    params = stringify_keys(params)
-    {secret, params} = strip_value(params, "secret")
+    stringify_params = stringify_keys(params)
+    
+    #tuple {secret, params}
+    secret_params = strip_value(stringify_params, "secret")
```

Comment:
Let's go back to using pattern matching here, it's not clear to me why it was removed in the first place.  In fact the resulting code is worse without it, we're now reliant on using pipe chains within function calls to get the values we want.